### PR TITLE
Migration endpoint to move ETH accounts to keys namespace

### PR DESCRIPTION
**Pull request by darioAnongba**
Migration endpoint to move ETH accounts to keys namespace

### CODE REVIEW

**Comment by dolanor** [comment]
Should the intellij files be there?

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<component name="InspectionProjectProfileManager">
```
Comment (ggarri) [inline .idea/inspectionProfiles/Project_Default.xml:1]: Please, undo your local `.idea` folder

File: src/service/migrations/eth-to-keys.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/consensys/quorum-hashicorp-vault-plugin/src/service/formatters"
+	"github.com/consensys/quorum-hashicorp-vault-plugin/src/utils"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func (c *controller) NewEthereumToKeysOperation() *framework.PathOperation {
+	return &framework.PathOperation{
+		Callback:    c.ethToKeysHandler(),
+		Summary:     "Migrates the current Ethereum accounts to the keys namespace",
```
Comment (ggarri) [inline src/service/migrations/eth-to-keys.go:16]: Minor: I think you can skip "the current" from the summary as it is self-explanatory already your are moving Ethereum Accounts

File: src/vault/use-cases/migrations/eth_to_keys.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		return err
+	}
+
+	status := &entities.MigrationStatus{
+		Status:    "pending",
+		StartTime: time.Now(),
+	}
+	uc.status[namespace] = status
+
+	go func() {
```
Comment (ggarri) [inline src/vault/use-cases/migrations/eth_to_keys.go:63]: IMO making an UC runs in async within go-routine it is a decision to be made on the caller so you might want to run the same UC synchonously for your integration-test for instance or whatever other cases.

File: src/vault/entities/migrations.go [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package entities
+
+import "time"
+
+type MigrationStatus struct {
```
Comment (ggarri) [inline src/vault/entities/migrations.go:5]: In order to provide better insides about what it is the actual status you should also add the total amount of accounts or just add the pending amount which is decresing over time

File: .idea/inspectionProfiles/Project_Default.xml (hunk 1 above)
Comment (darioAnongba) [reply .idea/inspectionProfiles/Project_Default.xml]: done

File: src/service/migrations/eth-to-keys.go (hunk 2 above)
Comment (darioAnongba) [reply src/service/migrations/eth-to-keys.go:16]: done

File: src/vault/entities/migrations.go (hunk 4 above)
Comment (darioAnongba) [reply src/vault/entities/migrations.go:5]: done

File: src/vault/use-cases/migrations/eth_to_keys.go [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+	if err != nil {
+		return err
+	}
+
+	status := &entities.MigrationStatus{
+		Status:    "pending",
+		StartTime: time.Now(),
+		Total:     len(addresses),
+	}
+	uc.status[namespace] = status
```
Comment (ggarri) [inline src/vault/use-cases/migrations/eth_to_keys.go:62]: You should lock the access to the map just before here

File: src/service/migrations/eth-to-keys.go [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+		},
+	}
+}
+
+func (c *controller) ethToKeysHandler() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+		namespace := formatters.GetRequestNamespace(req)
+
+		ctx = log.Context(ctx, c.logger)
+		err := c.useCases.EthereumToKeys().WithStorage(req.Storage).Execute(ctx, namespace)
```
Comment (ggarri) [inline src/service/migrations/eth-to-keys.go:54]: YOu should prevent users are not calling this EP twice. You could check for any existing status before calling the migration UC

**Review by ggarri — APPROVED** [review]

**Pull request merged**